File: templates/Uwp/Pages/ContentGrid/ViewModels/ContentGridViewDetailViewModel.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using Param_RootNamespace.Core.Models;
using Param_RootNamespace.Core.Services;

namespace Param_RootNamespace.ViewModels
{
    public class ContentGridViewDetailViewModel : System.ComponentModel.INotifyPropertyChanged
    {
        private SampleOrder _item;

        public SampleOrder Item
        {
            get { return _item; }
            set { Set(ref _item, value); }
        }

        public ContentGridViewDetailViewModel()
        {
        }

        public async Task InitializeAsync(long orderID)
        {
            var data = await SampleDataService.GetContentGridDataAsync();
            Item = data.First(i => i.OrderID == orderID);
        }
    }
}

File: templates/Uwp/Pages/ContentGrid._VB/ViewModels/ContentGridViewDetailViewModel.vb
Imports Param_RootNamespace.Core.Models
Imports Param_RootNamespace.Core.Services

Namespace ViewModels
    Public Class ContentGridViewDetailViewModel
        Inherits System.ComponentModel.INotifyPropertyChanged

        Private _item As SampleOrder

        Public Property Item As SampleOrder
            Get
                Return _item
            End Get
            Set(value As SampleOrder)
                [Set](_item, value)
            End Set
        End Property

        Public Sub New()
        End Sub

        Public Async Function InitializeAsync(orderId As Long) As Task
            Dim data = Await SampleDataService.GetContentGridDataAsync()
            Item = data.First(Function(i) i.OrderId = orderId)
        End Function

    End Class
End Namespace
